subsys/net/lib/openthread/platform/radio.c

```diff
@@ -36,10 +36,16 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #include "platform-zephyr.h"
 
 #define FCS_SIZE 2
+#define ACK_PKT_LENGTH 3
+
+#define FRAME_TYPE_MASK 0x07
+#define FRAME_TYPE_ACK 0x02
 
 static otRadioState sState = OT_RADIO_STATE_DISABLED;
 
 static otRadioFrame sTransmitFrame;
+static otRadioFrame ack_frame;
```

Comment:
How is this static buffer going to work?

Reply:
It just holds a copy of the ACK frame until it's processed by OpenThread.

Reply:
> It just holds a copy of the ACK frame until it's processed by OpenThread.

So, is this processing is 100% cooperative multitasking driven, even in the case of multiple OT radios? That's the only scenario when a static buffer would work, and I'd be surprised to learn that such processing is used on top of otherwise preemptively multitasked Zephyr (and such support would require a lot of checks).

Reply:
Only a single OT radio instance is supported. If ACK was requested, the radio tx call will block until the ACK is received or a timeout occurs. And I assumed here that radio driver is smart enough not to notify another ACK until another transmit is called (that would need some extra care). Do we need anything else here?

Reply:
> Do we need anything else here?

No, just wanted to be sure that these points were raised in the review and some responses given. Let me approve from my side.